aws_lambda_builders/workflow.py

```diff
@@ -229,7 +257,18 @@ def __init__(
         self.architecture = architecture
         self.is_building_layer = is_building_layer
         self.experimental_flags = experimental_flags if experimental_flags else []
+
         self.build_in_source = build_in_source
+        if build_in_source is None:
+            self.build_in_source = self.BUILD_IN_SOURCE_BY_DEFAULT
```

Comment:
confused here, we set the same logic both if and in elif?

`self.build_in_source = self.BUILD_IN_SOURCE_BY_DEFAULT`

Reply:
Basically, we should use the default value if the passed in value either <is none> or <is not supported> (but also we only show a warning message when it's not supported). Otherwise it should use the passed in value.

Reply:
```
if build_in_source not in self.BUILD_IN_SOURCE_SUPPORT.value:
   self.build_in_source = self.BUILD_IN_SOURCE_BY_DEFAULT
```

you could warnings for None too, since technically they do not need to be defined at the workflow level.

Does this suffice?

Reply:
@sriram-mv Not sure if I understood your comment.

So, most of the time `build_in_source` will likely be `None` as this will be the default from the SAM CLI (can't really set a True/False default since each workflow currently has a different value as default). On the lambda builders side, each workflow will decide what the default value is when it receives `None` (what the couple lines above accomplish).

I don't think we should warn for `None` because that means by default, if the customer doesn't provide any specific True/False value from the sam cli side, they'll get this warning message. I don't feel like it's a good experience to get a new warning message for a valid use case that they were already doing before. I think ideally we only show this warning message if they explicitly provide an unsupported value (which is what I currently have here).

Reply:
Having the same assignment in both if and elif makes me question if we could do a different logic here, which is why I proposed that change. Agree on not adding warning logs for something that has not changed.

Reply:
refactored a bit